make-release.sh

```diff
@@ -121,25 +121,28 @@ make generate_deployment
 make docker-build
 make docker-push
 
-# tag the release	
-git tag "${VERSION}"	
-git push origin "${VERSION}"	
-COMMIT_MSG="[release] Release ${VERSION}"	
-git commit -asm "${COMMIT_MSG}"	
-
-# now update ${BASEBRANCH} to the new snapshot version	
-git checkout "${BASEBRANCH}"	
-
-# change VERSION file + commit change into ${BASEBRANCH} branch	
-if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then	
-  # bump the y digit, if it is a major release	
-  [[ $BRANCH =~ ^([0-9]+)\.([0-9]+)\.x ]] && BASE=${BASH_REMATCH[1]}; NEXT=${BASH_REMATCH[2]}; (( NEXT=NEXT+1 )) # for BRANCH=7.27.x, get BASE=7, NEXT=28		
-  NEXT_VERSION_Y="${BASE}.${NEXT}.0-SNAPSHOT"	
-  bump_version "${NEXT_VERSION_Y}" "${BASEBRANCH}"	
-fi	
-# bump the z digit	
-[[ ${VERSION#v} =~ ^([0-9]+)\.([0-9]+)\.([0-9]+) ]] && BASE="${BASH_REMATCH[1]}.${BASH_REMATCH[2]}"; NEXT="${BASH_REMATCH[3]}"; (( NEXT=NEXT+1 )) # for VERSION=7.27.1, get BASE=7.27, NEXT=2
-NEXT_VERSION_Z="${BASE}.${NEXT}-SNAPSHOT"	
+# tag the release if the VERSION file has changed
+if [[ ! -z $(git status -s) ]]; then # dirty
+  COMMIT_MSG="[release] Release ${VERSION}"
+  git add VERSION
+  git commit -asm "${COMMIT_MSG}"
+  git tag "${VERSION}"
+  git push origin "${VERSION}"
+fi
+
+# now update ${BASEBRANCH} to the new snapshot version
+git checkout "${BASEBRANCH}"
+
+# change VERSION file + commit change into ${BASEBRANCH} branch
+if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
+  # bump the y digit, if it is a major release
+  [[ $BRANCH =~ ^([0-9]+)\.([0-9]+)\.x ]] && BASE=${BASH_REMATCH[1]}; NEXT=${BASH_REMATCH[2]}; (( NEXT=NEXT+1 )) # for BRANCH=0.1.x, get BASE=0, NEXT=2
```

Comment:
unlike original DWO, this one uses same versioning as Che (with v prefix for tag)

Reply:
so... what are you suggesting needs to change here? you can edit my PR directly :)

Reply:
Restore the original sample values you are trying to replace 😄 
```suggestion
  [[ $BRANCH =~ ^([0-9]+)\.([0-9]+)\.x ]] && BASE=${BASH_REMATCH[1]}; NEXT=${BASH_REMATCH[2]}; (( NEXT=NEXT+1 )) # for BRANCH=7.27.x, get BASE=7, NEXT=28
```